assets/app/scripts/services/auth.js

```diff
@@ -276,4 +276,18 @@ angular.module('openshiftConsole')
       }
     }
   };
-}]);
+}])
+
+/* A WebSocket factory for kubernetesContainerTerminal */
+.factory("AuthContainerWebSocket", function(API_CFG, AuthService) {
```

Comment:
Allow the web socket factory to return a web socket or the promise of a web socket, and no complexity is added to other projects. We should allow for the case where work is needed to go get tokens, etc, to establish the web socket call.